catalog/transformer/catalog-transformer-zip/src/main/java/org/codice/ddf/catalog/transformer/zip/ZipCompression.java

```diff
@@ -225,6 +219,50 @@ private Resource getResource(Metacard metacard) {
     return resource;
   }
 
+  private void writeResourceToZip(
+      ZipOutputStream zipOutputStream, String filename, Resource resource) {
+
+    try {
+      ZipEntry zipEntry = new ZipEntry(filename);
+      zipOutputStream.putNextEntry(zipEntry);
+
+      InputStream inputStream = resource.getInputStream();
+
+      writeDataToZip(zipOutputStream, inputStream);
+      inputStream.close();
+    } catch (IOException e) {
+      LOGGER.debug("Failed to add resource with id {} to zip.", resource.getName(), e);
+    }
+  }
+
+  private void writeDataToZip(ZipOutputStream zipOutputStream, InputStream inputStream)
+      throws IOException {
+    final byte[] bytes = new byte[1024];
+    int length;
+    while ((length = inputStream.read(bytes)) >= 0) {
+      zipOutputStream.write(bytes, 0, length);
+    }
+  }
+
+  private BinaryContent getBinaryContentFromZip(String filePath)
+      throws CatalogTransformerException {
+    BinaryContent binaryContent;
+    try {
+      File zipFile = new File(filePath);
+      InputStream fileInputStream = new ZipInputStream(new FileInputStream(zipFile));
+      binaryContent = new BinaryContentImpl(fileInputStream);
+      jarSigner.signJar(
+          zipFile,
+          System.getProperty(SystemBaseUrl.EXTERNAL_HOST),
```

Comment:
❗️ Properties should be retrieved via privileged actions.

Reply:
I'm thinking about throwing a CatalogTransformerException whenever a PrivilegedActionException occurs here which is a quick and easy fix. Would that be the best way of implementing this or is there a better cleaner solution that I'm overlooking?